src/allmydata/storage_client_http.py

```diff
@@ -0,0 +1,204 @@
+"""
+HTTP-based storage client.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from future.utils import PY2
+if PY2:
+    from future.builtins import filter, map, zip, ascii, chr, hex, input, next, oct, open, pow, round, super, bytes, dict, list, object, range, str, max, min  # noqa: F401
+
+import attr
+from zope.interface import implementer
+from twisted.internet.defer import inlineCallbacks, returnValue, Deferred, succeed
+
+from .interfaces import (
+    IStorageClientV2, IStorageServer,
+    TestVector, ReadVector, TestWriteVectors,
+    TestVectorOperator, WriteVector,
+)
+
+
+class _FakeRemoteReference(object):
+    """Emulate a Foolscap RemoteReference."""
+
+    def callRemote(self, action, *args, **kwargs):
+        return getattr(self, action)(*args, **kwargs)
+
+    def callRemoteOnly(self, action, *args, **kwargs):
+        getattr(self, action)(*args, **kwargs)
+        return None
```

Comment:
I don't know what uses `callRemoteOnly` but the real implementation silently discards all exceptions before returning `None`.

That's pretty no good ... but I'm not sure if allowing exceptions to propagate makes this a faithful emulation or not.

Reply:
Perhaps the real answer is "don't use callRemoteOnly in higher-level code". I'll file a ticket.